File: backend/app/models/party.py
from sqlalchemy import Column, String, Integer, Text
from app.models.base import BaseModel


class Party(BaseModel):
    __tablename__ = "parties"

    name = Column(String(100), unique=True, nullable=False, index=True)
    colour = Column(String(7), nullable=False)
    colour_secondary = Column(String(7), nullable=True)
    abbreviation = Column(String(10), nullable=True, unique=True)
    leader = Column(String(100), nullable=True)
    founded = Column(Integer, nullable=True)
    position = Column(String(50), nullable=True)
    description = Column(Text, nullable=True)
    summary = Column(Text, nullable=True)
    region = Column(String(50), nullable=True)
    priorities = Column(Text, nullable=True)

File: frontend/src/types/party.ts
export type Region = "England" | "Scotland" | "Wales" | "Northern Ireland"

export interface Party {
  id?: string
  slug: string
  name: string
  leader: string
  position: string
  summary: string
  description: string
  region: Region
  priorities: string[]
  founded: number
  color: string
  colourSecondary: string | null
  abbreviation: string
  seats?: number
  voteShare?: number
}

export interface PartyHistoryItem {
  year: number
  seats: number
  voteShare: number | null
  totalVotes: number
}

export interface PartyConstituencyItem {
  id: string
  name: string
  region: string
  winner: boolean
  majority: number | null
}
